impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/BeanConfiguratorImpl.java

```diff
@@ -311,7 +311,12 @@ public BeanConfiguratorImpl(Class<?> defaultBeanClass, BeanDeploymentFinder bean
     }
 
     public Bean<T> complete() {
-        return new ImmutableBean<>(this);
+        ImmutableBean<T> result = new ImmutableBean<>(this);
+        if (createCallback == null) {
```

Comment:
Well, you could make `BeanAttributesConfiguratorImpl.qualifiers` and `types` package-private and use something like:

````
throw BeanLogger.LOG.noCallbackSpecifiedForCustomBean("Configurator [types: " + Formats.formatTypes(attributes.types) + ", qualifiers: " + Formats.formatAnnotations(attributes.qualifiers) + "]");
````

Hm, `BeanAttributesConfiguratorImpl.types` should be final - pls add this change too.